src/phpDocumentor/ApiReference/Factory.php

```diff
@@ -47,25 +62,15 @@ public function create(DocumentGroupDefinitionInterface $definition)
             throw new InvalidArgumentException('Definition must be an instance of ' . DocumentGroupDefinition::class);
         }
 
-        // TODO: move this to a custom factory so that we can inject the factory in this class and then have the
-        // middleware as dependency of that factory.
+        $strategies = $this->stategies;
+        $strategies[] = new File(
```

Comment:
I would like tho implement this. But I will have to change the ProjectFactory class for this. It should be able to expose it's strategies. (Which I don't like). Or it should accept a FileStrategyAdapter on it's create method. Which I don't like either. Since the FileStrategyAdapter is very specific to the FileStrategy. 

The actual problem lies in the requirement of the strategy to get the filesystem from the definition. Which is only available when the Factory::create method is called....